src/theme/BlogPostItem/index.js

```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import { DiscussionEmbed } from 'disqus-react'
+import BlogPostItem from '@theme-original/BlogPostItem';
+import {useBlogPost} from '@docusaurus/plugin-content-blog/client'
+
+export default function BlogPostItemWrapper(props) {
+  const { metadata } = useBlogPost()
+  const { frontMatter, slug, title } = metadata
```

Comment:
I'm not sure if there's `slug` in `metadata` in the current version of Docusaurus. [Here](https://github.com/facebook/docusaurus/blob/dd4fe63a3dfbdbfc264ef3f6bb93dbae6434d24b/packages/docusaurus-plugin-content-blog/src/plugin-content-blog.d.ts#L245), I think, is the metadata's type definition. There's [`slug` in front matter](https://github.com/facebook/docusaurus/blob/dd4fe63a3dfbdbfc264ef3f6bb93dbae6434d24b/packages/docusaurus-plugin-content-blog/src/plugin-content-blog.d.ts#L157), but it's optional.

Disqus will use the page's URL if the [identifier](https://help.disqus.com/en/articles/1717084-javascript-configuration-variables#thispageidentifier) is undefined, so it's probably okay to simply omit the identifier and rely on blog's URLs being stable.